crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -268,6 +268,10 @@ use std::{cell::UnsafeCell, marker::PhantomData};
 ///
 /// [`Query`]: crate::system::Query
 /// [`ReadOnly`]: Self::ReadOnly
+#[diagnostic::on_unimplemented(
+    message = "`{Self}` is not valid `Query` data",
```

Comment:
```suggestion
    message = "`{Self}` is not valid to request as a `Query`'s data",
```

Reply:
"not valid to request as data in a `Query`" sounds more natural than "not valid to request as a `Query`'s data", in my opinion.

Reply:
Going with the `message` _"`{Self}` is not valid to request as data in a `Query`"_